### ISSUE TITLE
Possible memory leak in cg_rocket_datasource.cpp

### ISSUE THREAD

**Issue by Dimitriio** [Unvanquished/Unvanquished#1148]
I was reading some code [in cg_rocket_datasource.cpp](https://github.com/Unvanquished/Unvanquished/blob/5198bace54ed077114d1478ace479fc153bc32ff/src/cgame/cg_rocket_datasource.cpp#L423) I saw this method :
```
void CG_Rocket_CleanUpServerList( const char *table )
{
	int i;
	int j;
	int netSrc = CG_StringToNetSource( table );

	for ( i = AS_LOCAL; i <= AS_FAVORITES; ++i )
	{
		if ( !table || !*table || i == netSrc )
		{
			for ( j = 0; j < rocketInfo.data.serverCount[ i ]; ++j )
			{
				BG_Free( rocketInfo.data.servers[ i ][ j ].name );
				BG_Free( rocketInfo.data.servers[ i ][ j ].label );
				BG_Free( rocketInfo.data.servers[ i ][ j ].addr );
				BG_Free( rocketInfo.data.servers[ i ][ j ].mapName );
				rocketInfo.data.serverCount[ i ] = 0;
			}
		}
	}
}
```

Should the line `rocketInfo.data.serverCount[ i ] = 0;` be oustide the "for loop" like that :
```
void CG_Rocket_CleanUpServerList( const char *table )
{
	int i;
	int j;
	int netSrc = CG_StringToNetSource( table );

	for ( i = AS_LOCAL; i <= AS_FAVORITES; ++i )
	{
		if ( !table || !*table || i == netSrc )
		{
			for ( j = 0; j < rocketInfo.data.serverCount[ i ]; ++j )
			{
				BG_Free( rocketInfo.data.servers[ i ][ j ].name );
				BG_Free( rocketInfo.data.servers[ i ][ j ].label );
				BG_Free( rocketInfo.data.servers[ i ][ j ].addr );
				BG_Free( rocketInfo.data.servers[ i ][ j ].mapName );
			}
			rocketInfo.data.serverCount[ i ] = 0;
		}
	}
}
```
I feel that for now we just iterate on the first element of the list which create a small memory leak.

**Comment by DolceTriade** [comment]
You are correct.  If you would like to submit a PR feel free, otherwise we will fix it.

**Comment by illwieckz** [comment]
fixed by #1149
